contracts/staking/test/pools_test.ts

```diff
@@ -82,126 +110,129 @@ blockchainTests('Staking Pool Management', env => {
                 new MakerActor(makerAddresses[2], stakingWrapper),
             ];
             // create pool
-            const poolId = await poolOperator.createStakingPoolAsync(operatorShare);
+            const poolId = await poolOperator.createStakingPoolAsync(operatorShare, true);
             expect(poolId).to.be.equal(stakingConstants.INITIAL_POOL_ID);
             // add makers to pool
-            const makerApprovals = [
-                makers[0].signApprovalForStakingPool(poolId),
-                makers[1].signApprovalForStakingPool(poolId),
-                makers[2].signApprovalForStakingPool(poolId),
-            ];
-            await poolOperator.addMakerToStakingPoolAsync(poolId, makerAddresses[0], makerApprovals[0].signature);
-            await poolOperator.addMakerToStakingPoolAsync(poolId, makerAddresses[1], makerApprovals[1].signature);
-            await poolOperator.addMakerToStakingPoolAsync(poolId, makerAddresses[2], makerApprovals[2].signature);
+            await Promise.all([
```

Comment:
I wonder if it would be useful to introduce a `MakerMasterActor` that takes has a function called "assignMakerActions" that takes a method name, and then calls this method with the `poolId` and encapsulates some of this logic. Thoughts?

I'm starting to see the `Promise.all` used a fair amount, so it might be good to consider a higher level abstraction.

Reply:
Idk, a `MakerMasterActor` doesn't really map to a real-world entity though. But we could introduce `batchAddMakerToStakingPoolAsync` and `batchRemoveMakerFromStakingPoolAsync` to `PoolOperatorActor` though